src/legacy/ui/public/vis/components/tooltip/_collect_branch.js

```diff
@@ -17,26 +17,27 @@
  * under the License.
  */
 
-// eslint-disable-next-line import/no-default-export
-export default function (leaf) {
+export function collectBranch(leaf) {
   // walk up the branch for each parent
   function walk(item, memo) {
     // record the the depth
     const depth = item.depth - 1;
 
-    // Using the aggConfig determine what the field name is. If the aggConfig
+    // Using the column name determine what the field name is. If the name
     // doesn't exist (which means it's an _all agg) then use the level for
     // the field name
-    const { aggConfig } = item;
-    const field = (aggConfig && aggConfig.makeLabel())
-      || (aggConfig && aggConfig.label)
-      || ('level ' + item.depth);
+    function getFieldName(i) {
+      if (i.rawData && i.rawData.column > -1) {
+        const { column, table } = i.rawData;
+        return table.columns[column].name;
+      }
+      return i.name || `level ${i.depth}`;
```

Comment:
`i.name` is needed here so we can get the `"Count"` label back for metrics, but TBH I'm not sure when/if we should ever expect to get the fallback `"level n"` text. AFAICT that code should be unreachable.